docs/source/en/model_doc/audioflamingo3.md

```diff
@@ -0,0 +1,393 @@
+
+print(transcription[0])
+```
+
+## How the model works
+
+### Architecture
+
+* **AudioFlamingo3Encoder**
+  Whisper-style feature extractor + encoder → average-pool over time (stride 2) → LayerNorm.
+  Produces per-frame hidden states at the post-pool rate.
+
+* **AudioFlamingo3MultiModalProjector**
+  A small MLP that maps encoder features to the language model’s hidden size.
+
+* **AudioFlamingo3ForConditionalGeneration**
+  A causal language model that accepts text embeddings where each `<sound>` token slot is replaced, in place, by an audio frame embedding. No sequence-length change is introduced by fusion.
+
+### Processor-level alignment
+
+1. Each raw waveform is split into fixed-length windows based on the feature extractor’s `chunk_length` (seconds) and `sampling_rate` (Hz).
+2. For each window, the processor computes the number of post-pool frames `post_pool_len` that the encoder will output (matching the conv/pool schedule).
+3. The processor expands the audio placeholder token by the total number of post-pool frames across all windows.
+
+   * If the prompt contains no `<sound>`, all expanded tokens are inserted at the start of the user message in the chat template (or prepended to plain text if no template is used).
+   * If the prompt contains `<sound>`, exactly one placeholder is supported and it is expanded in place. Multiple placeholders are not supported.
+4. The model later replaces those token positions with the corresponding projected audio embeddings.
+
+This design guarantees a 1:1 match between placeholder positions in the text and encoder frame outputs, enabling safe batching without ragged audio–text concatenation.
+
+## Usage patterns
+
+### Transcription shortcut
+
+For automatic speech recognition you can skip writing the default instruction each time and call
+[`~transformers.AudioFlamingo3Processor.apply_transcription_request`]:
+
+```python
+inputs = processor.apply_transcription_request(audio=audio_array)
+```
+
+Pass `prompt="Transcribe the input speech."` (or a list of prompts for batched audio) to customize the instruction while
+keeping the audio placeholder handling.
+
+`audio` accepts in-memory arrays, local file paths, or URLs. Any processor kwargs (`text_kwargs`, `audio_kwargs`, etc.)
+are forwarded, so you can tweak padding or tensor formats just like when calling `processor(...)`.
+
+### Single-turn prompts (recommended)
+
+You can omit `<sound>` entirely and let the processor insert the expanded tokens:
+
+```python
+prompt = "Describe the melody in detail."
+inputs = processor(text=prompt, audio=audio_array)
+```
+
+### Explicit placeholder control (advanced)
+
+If you include `<sound>` in your prompt, provide exactly one placeholder; it will be expanded to match the total number of post‑pool frames across all windows:
+
+```python
+# For a clip that may span multiple windows:
+prompt = "<sound>\nDescribe the audio in detail."
+inputs = processor(text=prompt, audio=audio_array)
+```
```

Comment:
@lashahub related to [this discussion](https://github.com/huggingface/transformers/pull/40290/files#r2486281860) which I continued offline with @eustlb , I'm going to remove support for such usage, as we prefer usage through the standardized chat template.